### PR TITLE
feat: add how to reduce memory usage while building

### PR DESCRIPTION
**Pull request by Shin-kyoto**
## Description

Add how to reduce memory usage while buildng.
I referred to documents below.

- https://answers.ros.org/question/368249/colcon-build-number-of-threads/
- https://colcon.readthedocs.io/en/released/reference/executor-arguments.html
- [Tech blog about autoware](https://medium.com/tier-iv-tech-blog/autoware%E3%81%AE%E3%83%81%E3%83%A5%E3%83%BC%E3%83%88%E3%83%AA%E3%82%A2%E3%83%AB%E3%81%A8component%E3%81%AE%E3%83%8F%E3%82%A4%E3%83%AC%E3%83%99%E3%83%AB%E3%81%AA%E8%A8%AD%E8%A8%88-67a8a2a1b9ba#:~:text=MAKEFLAGS%3D%2Dj1%20colcon%20build%20%2D%2Dcmake%2Dargs%20%2DDCMAKE_BUILD_TYPE%3DRelease%20%2D%2Dparallel%2Dworkers%201%20%23%20MAKEFLAGS%E3%81%AF%E3%83%91%E3%83%83%E3%82%B1%E3%83%BC%E3%82%B8%E5%86%85%E3%81%AE%E4%B8%A6%E5%88%97%E6%95%B0%E3%80%81parallel%2Dworkers%E3%81%AF%E3%83%91%E3%83%83%E3%82%B1%E3%83%BC%E3%82%B8%E3%81%AE%E4%B8%A6%E5%88%97%E6%95%B0)
- https://askubuntu.com/questions/1295713/make-use-of-j-option
- https://www.gnu.org/software/make/manual/make.html#Parallel-Disable

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The Reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: docs/support/troubleshooting.md [hunk 1]
```diff
@@ -86,7 +86,13 @@ A workaround here is to limit the job number while building.
 MAKEFLAGS="-j4" colcon build --symlink-install --cmake-args -DCMAKE_BUILD_TYPE=Release
 ```
 
-You can adjust `-j4` to any number based on your system.
+You can adjust `-j4` to any number based on your system. Please see [manual page of GNU make](https://www.gnu.org/software/make/manual/make.html#Parallel-Disable) for details.
```
Comment (kenji-miyake) [inline docs/support/troubleshooting.md:89]: [style]

```suggestion
You can adjust `-j4` to any number based on your system.
For more details, see the [manual page of GNU make](https://www.gnu.org/software/make/manual/make.html#Parallel-Disable).
```

File: docs/support/troubleshooting.md [hunk 2]
```diff
@@ -86,7 +86,13 @@ A workaround here is to limit the job number while building.
...
+
+By reducing the number of packages built in parallel, you can also reduce the amount of memory used. In the following example, the number of packages built in parallel is set to 1, and the number of jobs used by `make` is limited to 1.
```
Comment (kenji-miyake) [inline docs/support/troubleshooting.md:91]: [style]

```suggestion
By reducing the number of packages built in parallel, you can also reduce the amount of memory used.
In the following example, the number of packages built in parallel is set to 1, and the number of jobs used by `make` is limited to 1.
```

File: docs/support/troubleshooting.md (hunk 1 above)
Comment (Shin-kyoto) [reply docs/support/troubleshooting.md]: @kenji-miyake 
Thanks a lot!

File: docs/support/troubleshooting.md [hunk 3]
```diff
@@ -87,6 +87,14 @@ MAKEFLAGS="-j4" colcon build --symlink-install --cmake-args -DCMAKE_BUILD_TYPE=R
...
 
 You can adjust `-j4` to any number based on your system.
+For more details, see the [manual page of GNU make](https://www.gnu.org/software/make/manual/make.html#Parallel-Disable).
+
+By reducing the number of packages built in parallel, you can also reduce the amount of memory used.
+In the following example, the number of packages built in parallel is set to 1, and the number of jobs used by `make` is limited to 1.
+
+```bash
+MAKEFLAGS="-j1" colcon build --symlink-install --cmake-args -DCMAKE_BUILD_TYPE=Release --parallel-workers 1
+```
```
[the remark this answers is not in the document]
Comment (Shin-kyoto) [reply docs/support/troubleshooting.md:97]: @kenji-miyake 
Thank you for your comment.
I added note in https://github.com/autowarefoundation/autoware-documentation/pull/356/commits/dbc08ffb23b1b8a49f5563c2b4e25129010ac15a

**Pull request merged**